### PR TITLE
Add initial support for transforms to layout_2020

### PR DESCRIPTION
**Pull request by mrobinson**
---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [ ] These changes fix #___ (GitHub issue number if applicable)

- [x] There are tests for these changes OR
- [ ] These changes do not require tests because ___

### CODE REVIEW

File: components/layout_2020/display_list/stacking_context.rs [hunk 1]
```diff
@@ -476,6 +499,123 @@ impl BoxFragment {
             );
         }
     }
+
+    fn build_reference_frame_if_necessary(
+        &self,
+        builder: &mut DisplayListBuilder,
+        border_rect: &PhysicalRect<Length>,
+    ) -> bool {
```
Comment (SimonSapin) [inline components/layout_2020/display_list/stacking_context.rs]: Please add a doc-comment to explain what this return value means.

File: components/layout_2020/display_list/stacking_context.rs [hunk 2]
```diff
@@ -476,6 +499,123 @@ impl BoxFragment {
...
+        if !self.has_transform_or_perspective() {
+            return false;
+        }
+        let au_border_rect = f32_rect_to_au_rect(Rect::new(
```
Comment (SimonSapin) [inline components/layout_2020/display_list/stacking_context.rs]: Could we avoid using `Au` and stay within `Length` or `f32` coordinates in CSS `px` unit? (To help do that replace `to_used_value` with `percentage_relative_to`.) It looks like the transformation matrix is ultimately in floating point anyway.

File: components/layout_2020/display_list/stacking_context.rs (hunk 1 above)
Comment (mrobinson) [reply components/layout_2020/display_list/stacking_context.rs]: Okay.

File: components/layout_2020/display_list/stacking_context.rs (hunk 2 above)
Comment (mrobinson) [reply components/layout_2020/display_list/stacking_context.rs]: Thanks for the review. I've pushed a new version of the branch that addresses this comment by modifying the style code to accept Length values when resolving transformations.

**Comment by SimonSapin** [comment]
Looks good, thanks!

@bors-servo r+

**Pull request merged**